stac_validator.py

```diff
@@ -197,11 +172,37 @@ def is_valid_url(self, url):
         except:
             return False
 
-    async def run(self):
+    def fetch_and_parse_file(self, input_path):
         """
-        Entry point
-        :return: message json
+        Fetch and parse STAC file
+        :return: content or error message
         """
+        err_message = {}
+        data = None
+
+        try:
+            if self.is_valid_url(input_path):
+                resp = requests.get(input_path)
+                data = resp.json()
+            else:
+                with open(input_path) as f:
+                    data = json.load(f)
+
+        except JSONDecodeError as e:
+            err_message["valid_stac"] = False
+            err_message["error_type"] = "InvalidJSON"
+            err_message["error_message"] = f"{input_path} is not Valid JSON"
+
+        except FileNotFoundError as e:
+            err_message["valid_stac"] = False
+            err_message["error_type"] = "FileNotFoundError"
+            err_message["error_message"] = f"{input_path} cannot be found"
+
+        return data, err_message
+
+    def _validate(self, stac_path):
```

Comment:
`_validate` is a top level validation function which does: 
- downloads and parse file (call `fetch_and_parse_file`)
- checks if we are processing an item, a collection or a catalog 
- returns: 
   - error messages (download, parsing or validation)
   - status
   - children